File: onprom-ocel/src/main/java/it/unibz/inf/pm/ocel/examples/ImportXmlocelExportXmlocel.java
package it.unibz.inf.pm.ocel.examples;

import it.unibz.inf.pm.ocel.OcelInitial;
import org.dom4j.DocumentException;

import java.util.Map;

public class ImportXmlocelExportXmlocel {

    public static void main(String[] args) throws Exception {
        try {
            System.out.println("validated input:");
            OcelInitial.validate("ocel/logs/minimal.xmlocel", "ocel/schemas/schema.xml");
            Map log = (Map)OcelInitial.import_log("ocel/logs/minimal.xmlocel");
            OcelInitial.export_log(log,"ocel/examples/log3.xmlocel");

            System.out.println("\nvalidated output:");
            OcelInitial.validate("ocel/examples/log3.xmlocel", "ocel/schemas/schema.xml");
        } catch (DocumentException e) {
            e.printStackTrace();
        }
    }
}

File: pyocel/examples/import_xmlocel_export_xmlocel.py
import ocel


def execute_script():
    print("validated input", ocel.validate("../logs/minimal.xmlocel", "../schemas/schema.xml"))
    log = ocel.import_log("../logs/minimal.xmlocel")
    ocel.export_log(log, "log2.xmlocel")
    print("validated output", ocel.validate("log2.xmlocel", "../schemas/schema.xml"))
    ocel.import_log("log2.xmlocel")


if __name__ == "__main__":
    execute_script()
